server/plugin/api.go

```diff
@@ -1258,6 +1422,297 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
 	p.writeJSON(w, resp)
 }
 
+func (p *Plugin) updateIssue(c *UserContext, w http.ResponseWriter, r *http.Request) {
+	type IssueRequest struct {
+		Title       string   `json:"title"`
+		Body        string   `json:"body"`
+		Repo        string   `json:"repo"`
+		PostID      string   `json:"post_id"`
+		ChannelID   string   `json:"channel_id"`
+		Labels      []string `json:"labels"`
+		Assignees   []string `json:"assignees"`
+		Milestone   int      `json:"milestone"`
+		IssueNumber int      `json:"issue_number"`
+	}
+
+	// get data for the issue from the request body and fill IssueRequest object
+	issue := &IssueRequest{}
+	if err := json.NewDecoder(r.Body).Decode(&issue); err != nil {
+		c.Log.WithError(err).Warnf("Error decoding JSON body")
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "Please provide a JSON object.", StatusCode: http.StatusBadRequest})
```

Comment:
i think this error message is not correct as the input can be a json and we can still get this error